src/community/jdbcstore/src/main/java/org/geoserver/jdbcstore/internal/JDBCQueryHelper.java

```diff
@@ -0,0 +1,568 @@
+        builder.append(") VALUES ( ");
+        {
+            int i=0;
+            for(Assignment<?> assign:assignments) {
+                if(i++>0) builder.append(", ");
+                builder.append("?");
+                assign.addAsParameter(builder);
+            }
+        }
+        builder.append(");");
+        
+        List<Integer> result = anyInsertQuery(builder);
+        assert(result == null || result.size() == 1);
+        return result==null ? null : result.get(0);
+    }
+        
+    public List<Map<String, Object>> anyMultiQuery(QueryBuilder query, Field<?>... fields) {
+        try (Connection c = ds.getConnection()) {
+            LOGGER.log(Level.FINEST, query.toString());
+            PreparedStatement stmt = query.toStatement(c);
+            
+            try {
+                ResultSet rs = stmt.executeQuery();
+                
+                try {
+                    ArrayList<Map<String, Object>> results = new ArrayList<Map<String, Object>>();
+                    while(rs.next()) {           
+                        Map<String, Object> result = new HashMap<String, Object>();
+                        for(int i = 0; i< fields.length; i++) {
+                            result.put(fields[i].getFieldName(), fields[i].getValue(rs));
+                        }
+                        results.add(result);
+                    } 
+                    return results;
+                } finally {
+                    rs.close();
+                }
+            } finally {
+                stmt.close();
+            }
+        } catch (SQLException ex) {
+            throw new IllegalStateException("MultiSelectQuery Failed",ex);
+        }
+    }
+    
+    public Map<String, Object> anyQuery(QueryBuilder query, Field<?>... fields) {
+        try (Connection c = ds.getConnection()) {
+            LOGGER.log(Level.FINEST, query.toString());
+            PreparedStatement stmt = query.toStatement(c);
+            
+            try {            
+                ResultSet rs = stmt.executeQuery();
+                
+                try {
+                    if(rs.next()) {                
+                        assert(rs.last());
+                        Map<String, Object> result = new HashMap<String, Object>();
+                        for(int i = 0; i< fields.length; i++) {
+                            result.put(fields[i].getFieldName(), fields[i].getValue(rs));
+                        }
+                        return result;
+                    } else {
+                        return null;
+                    }
+                } finally {
+                    rs.close();
+                }
+            } finally {
+                stmt.close();
+            }
+        } catch (SQLException ex) {
+            throw new IllegalStateException("SelectQuery Failed",ex);
+        }
+    }
+    
+    public int anyUpdateQuery(QueryBuilder query) {
+        try (Connection c = ds.getConnection()) {
```

Comment:
try-with-resources; you could move the logging into the body which is a slightly different ordering, or before as it does not need the resources.